### ISSUE TITLE
dlib fails to build

### ISSUE THREAD

**Issue by mtrsk** [nix-community/poetry2nix#35]
I'm currently switching some poetry projects to `poetry2nix`, it worked great so far with exception of the [following dependency](https://pypi.org/project/face_recognition/), which gets broken due to having [dlib](https://pypi.org/project/dlib/) as a dependency. 

Here's a minimal example `pyproject.toml`:

```
[tool.poetry]
name = "example"
version = "0.1.0"
description = ""
authors = [""]

[tool.poetry.dependencies]
python = "^3.7"
fastapi = "^0.46.0"

[tool.poetry.dev-dependencies]
pytest = "^5.2"

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```
and `shell.nix`:
```
{ bootstrap ? import <nixpkgs> {}
, pkgs ? import (bootstrap.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs-channels";
    rev = "100012e55bc2a82fc680cba31a426ad38ead6fab";
    sha256 = "0cvnx852n0krci9hi1rpcldx1kcpmvi5ihf2awvwfayvzp4wic8z";
  }) {}
}:

let
  pythonEnv = pkgs.poetry2nix.mkPoetryEnv {
    python = pkgs.python37;
    poetrylock = ./poetry.lock;
  };
  systemDeps = with pkgs; [
    poetry
  ];
in
pkgs.mkShell {
  buildInputs = systemDeps ++ [ pythonEnv ];
}
```
Works fine. Now, adding `face_recognition` to `pyproject.toml`:
```
[tool.poetry.dependencies]
python = "^3.7"
fastapi = "^0.46.0"
face_recognition = "^1.2.3"
```
gives:
```
$ nix-shell -v
(...)
RuntimeError:
*******************************************************************
 CMake must be installed to build the following extensions: dlib
*******************************************************************

builder for '/nix/store/mrcdzqmpvc3s28033alxqklymj68l237-python3.7-dlib-19.19.0.drv' failed with exit code 1
cannot build derivation '/nix/store/ladrji9svq8kb0a2mhamvw8j70jb8zcj-python3-3.7.6-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/ladrji9svq8kb0a2mhamvw8j70jb8zcj-python3-3.7.6-env.drv' failed
```
* First attempt: Overriding dlib
```
{ bootstrap ? import <nixpkgs> {}
, pkgs ? import (bootstrap.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs-channels";
    rev = "100012e55bc2a82fc680cba31a426ad38ead6fab";
    sha256 = "0cvnx852n0krci9hi1rpcldx1kcpmvi5ihf2awvwfayvzp4wic8z";
  }) {}
}:

let
  pythonEnv = pkgs.poetry2nix.mkPoetryEnv {
    python = pkgs.python37;
    poetrylock = ./poetry.lock;
    overrides = [
      pkgs.poetry2nix.defaultPoetryOverrides
      (self: super: {
        dlib = super.dlib.overrideAttrs(old: {
          nativeBuildInputs = [ pkgs.cmake pkgs.pkgconfig ];
          buildInputs = [ pkgs.libpng pkgs.libjpeg ];
        });
      })
    ];
  };
  systemDeps = with pkgs; [
    poetry
  ];
in
pkgs.mkShell {
  buildInputs = systemDeps ++ [ pythonEnv ];
}
```
Produces:
```
$ nix-shell -v
(...)
copying coverage/fullcoverage/encodings.py -> build/lib.linux-x86_64-3.7/coverage/fullcoverage
running build_ext
building 'coverage.tracer' extension
creating build/temp.linux-x86_64-3.7
creating build/temp.linux-x86_64-3.7/coverage
creating build/temp.linux-x86_64-3.7/coverage/ctracer
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/nix/store/bzvpdbzdy3a1qba8f9ynrzyz811c10pp-python3-3.7.6/include/python3.7m -c coverage/ctracer/datastack.c -o build/temp.linux-x86_64-3.7/coverage/ctracer/datastack.o
builder for '/nix/store/fc40iwcg8hhpnw2n26qprlps7zc25jpd-python3.7-dlib-19.19.0.drv' failed with exit code 1
cannot build derivation '/nix/store/zbr6rixm6nrshzi7149zbzy25dawjz2d-python3-3.7.6-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/zbr6rixm6nrshzi7149zbzy25dawjz2d-python3-3.7.6-env.drv' failed
```

Any hints on how to solve this issue with dlib?

**Comment by adisbladis** [comment]
Fixed in 380a8aa482211679f36462edac736d6b8038335d.

**Comment by mtrsk** [comment]
Thanks for fixing it, this project is amazing btw!
